File: patterns/audio/openai-like/javascript/transcribe.ts
import 'dotenv/config';
import fs from 'fs';
import OpenAI from 'openai';

/**
 * Demonstrates Speech-to-Text (transcription) using OpenAI's Whisper model.
 */
async function main() {
    const apiKey = process.env.OPENAI_API_KEY;
    if (!apiKey) {
        console.error('Please set OPENAI_API_KEY.');
        return;
    }

    const client = new OpenAI({ apiKey });

    const audioFilePath = 'speech.mp3';

    if (!fs.existsSync(audioFilePath)) {
        console.log(`Note: '${audioFilePath}' not found. Please place an audio file to run this.`);
        console.log('Code structure is provided for reference.\n');
        return;
    }

    console.log('Transcribing audio...');

    try {
        const transcription = await client.audio.transcriptions.create({
            file: fs.createReadStream(audioFilePath),
            model: 'whisper-1',
        });

        console.log('\nTranscription:');
        console.log(transcription.text);
    } catch (error) {
        console.error('Error:', error);
    }
}

main();

File: patterns/audio/openai-like/python/transcribe.py
import os
from openai import OpenAI
from dotenv import load_dotenv

load_dotenv()

def main():
    """
    Demonstrates Speech-to-Text (transcription) using OpenAI's Whisper model.
    """
    api_key = os.environ.get("OPENAI_API_KEY")
    if not api_key:
        print("Please set OPENAI_API_KEY.")
        return

    client = OpenAI(api_key=api_key)

    # For this example, we assume you have a local audio file.
    # If the file doesn't exist, we will just print what the code WOULD do.
    audio_file_path = "speech.mp3"

    if not os.path.exists(audio_file_path):
        print(f"Note: '{audio_file_path}' not found. Please place an audio file to run this.")
        print("Code structure is provided below for reference.\n")
        return

    print("Transcribing audio...")
    
    with open(audio_file_path, "rb") as audio_file:
        transcription = client.audio.transcriptions.create(
            model="whisper-1",
            file=audio_file,
            response_format="text"
        )
    
    print("\nTranscription:")
    print(transcription)

if __name__ == "__main__":
    main()
